<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="img_topics" default="[]"/>
    <arg name="output_topic" default="height_map"/>
    <arg name="camera_info_topics" default="[]"/>
    <arg name="hm_frame" default="base_link"/>
    <arg name="lss_config_path" default="$(dirname)/../../monoforce/config/lss_cfg.yaml"/>
    <arg name="weights" default="$(dirname)/../../monoforce/config/weights/lss/lss.pt"/>
    <arg name="max_msgs_delay" default="0.1"/>
    <arg name="max_age" default="0.2"/>
    <arg name="calib_path" default="''"/>
    <arg name="n_cams" default="''"/>

    <!-- Terrain encoder node (LSS) -->
    <node name="terrain_encoder" pkg="monoforce" type="terrain_encoder" output="screen">
        <rosparam subst_value="true">
            lss_config_path: $(arg lss_config_path)
            img_topics: $(arg img_topics)
            camera_info_topics: $(arg camera_info_topics)
            hm_frame: $(arg hm_frame)
            weights: $(arg weights)
            max_msgs_delay: $(arg max_msgs_delay)
            max_age: $(arg max_age)
            calib_path: $(arg calib_path)
            n_cams: $(arg n_cams)
        </rosparam>
        <remap from="height_map" to="$(arg output_topic)"/>
    </node>
</launch>
